src/amo/components/CollectionList/index.js

```diff
@@ -0,0 +1,174 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import {
+  expandCollections, fetchUserCollections,
+} from 'amo/reducers/collections';
+import { getCurrentUser } from 'amo/reducers/users';
+import AuthenticateButton from 'core/components/AuthenticateButton';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import Button from 'ui/components/Button';
+import Card from 'ui/components/Card';
+import type {
+  CollectionsState, CollectionType,
+} from 'amo/reducers/collections';
+import type { UsersStateType } from 'amo/reducers/users';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ReactRouterLocation } from 'core/types/router';
+
+import './styles.scss';
+
+
+export type Props = {|
+  collections: Array<CollectionType> | null,
+  currentUsername: string | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  isLoggedIn: boolean,
+  loadingUserCollections: boolean,
+  location: ReactRouterLocation,
+|};
+
+export class CollectionListBase extends React.Component<Props> {
+  static defaultProps = {
+    creating: false,
+  };
+
+  componentDidMount() {
+    const {
+      collections,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      loadingUserCollections,
+    } = this.props;
+
+    if (currentUsername && !loadingUserCollections && !collections) {
+      dispatch(fetchUserCollections({
+        errorHandlerId: errorHandler.id, username: currentUsername,
+      }));
+    }
+  }
+
+  render() {
+    const { i18n, isLoggedIn, location, collections } = this.props;
+    const myCollectionsHeader = i18n.gettext('My Collections');
+
+    if (!isLoggedIn) {
+      return (
+        <AuthenticateButton
+          noIcon
+          location={location}
+          logInText={i18n.gettext('Log in to view your collections')}
+        />
+      );
```

Comment:
It would be nice to reuse the `Card` on the left to display this button, because it looks a bit weird currently:

![screen shot 2018-06-22 at 14 17 52](https://user-images.githubusercontent.com/217628/41776222-3148bd6e-7627-11e8-8964-b4903e190088.png)